statsd/client_test.go

```diff
@@ -437,6 +445,118 @@ func TestEntityID(t *testing.T) {
 	}
 }
 
+func TestDDEnvServiceVersionSet(t *testing.T) {
+	esv := []string{ddEnvName, ddServiceName, ddVersionName}
+	for _, t := range esv {
+		initialValue, initiallySet := os.LookupEnv(t)
+		if initiallySet {
+			defer os.Setenv(t, initialValue)
+		} else {
+			defer os.Unsetenv(t)
+		}
+	}
+	os.Setenv(ddEnvName, "prod")
+	os.Setenv(ddServiceName, "dog")
+	os.Setenv(ddVersionName, "abc123")
+	client, err := statsd.New("localhost:8125")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(client.Tags) != 3 {
+		t.Errorf("Expecting three tags, got %d", len(client.Tags))
+	}
+	if client.Tags[0] != "env:prod" {
+		t.Errorf("Bad tag value, got %s", client.Tags[0])
+	}
+	if client.Tags[1] != "service:dog" {
+		t.Errorf("Bad tag value, got %s", client.Tags[1])
+	}
+	if client.Tags[2] != "version:abc123" {
+		t.Errorf("Bad tag value, got %s", client.Tags[2])
+	}
```

Comment:
We could test once with a list.

```go
assert.Equal(t, client.Tags, []string{"env:prod", "service:dog", "version:abc123"})
```

Reply:
Sure thing; I didn't see much use of `assert` in the other test cases but there's no reason not to use it here.